persistence/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.mapper.GiftCertificateMapper;
+import com.epam.esm.dao.query.DatabaseQuery;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.GiftCertificateQueryParameters;
+import com.epam.esm.entity.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The type GiftCertificateDaoImpl.
+ *
+ * @author Verbovskiy Sergei
+ * @version 1.0
+ */
+@Repository
+public class GiftCertificateDaoImpl implements GiftCertificateDao {
+    private final JdbcTemplate jdbcTemplate;
+    private final GiftCertificateMapper giftCertificateMapper = new GiftCertificateMapper();
+
+    @Autowired
+    public GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public GiftCertificate add(GiftCertificate giftCertificate) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement statement = connection.prepareStatement(DatabaseQuery.ADD_GIFT_CERTIFICATE,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, giftCertificate.getName());
```

Comment:
I would prefer to make the final fields for these positions

Reply:
> I would prefer to make the final fields for these positions

Thank you, fixed